### ISSUE TITLE
Conventions for a React app’s directory structure

### ISSUE THREAD

**Issue by jferrettiboke** [airbnb/javascript#1555]
I read a lot about how to structure a React app regarding to the app sizes. However, I think there are many approaches on the table although almost  all these approaches are reduced to "by types", "by features" or "by domain".

Structuring the app's directory files is very important as you know. So, please, I would like to see on [https://github.com/airbnb/javascript/tree/master/react](https://github.com/airbnb/javascript/tree/master/react) a point, explaining how to structure a React app with the best conventions in mind.
Do you think this would be useful and suitable to show it on "Airbnb React/JSX Style Guide"? If not, I would appreciate if someone could help, sharing his/her point of view on how Airbnb manages this or how to structure React apps and not having headache in the future about scaling.

Thanks a lot!

**Comment by ljharb** [comment]
I don't think that this is a good thing for a general style guide to specify.

A given app certainly should have a convention - but beyond "all tests should be in a top-level directory, as should all source files, and both should have a parallel directory structure", I'm not sure what would be a good idea to codify.

For example, Airbnb's main website repo is structured largely based on how the original Rails app structured it, which is largely dictated by whatever version of Rails the app began with. That works well for this app, but in no way should be the inspiration for other projects. Many of airbnb's open source projects are small, and have no directory structure - some are larger, and have a very project-specific structure.

In other words, while I think it's important to *have* a structure, I don't think it's important that this be the same from project to project - and additionally, I think it's important for it *not* to be the same from project to project, but rather to conform to what makes sense for each project.

**Comment by ljharb** [comment]
@JohnForster a PR would be welcome :-)
